File: src/main/java/Joueur.java
public abstract class Joueur {

    private String pseudo;
    private int cote;
    private int score;

    public Joueur(String pseudo, int cote) {
        this.pseudo = pseudo;
        this.cote = cote;
        this.score = 0;
    }

    public abstract int jouer(Plateau plateau);


    public void ajouterPoint(int point) {
        this.score += point;
    }

    public int getScore() {
        return score;
    }

    public String getPseudo() {
        return pseudo;
    }

    public void setPseudo(String pseudo) {
        this.pseudo = pseudo;
    }

    public void setCote(int cote) {
        this.cote = cote;
    }

    public String toString(){
        return "Le joueur " + this.pseudo + " a " + this.score + " points.";
    }
}

File: Awale/Joueur.ts
import { Plateau } from 'plateau';

export abstract class Joueur {
	
	private pseudo: String;
	private cote: number;
	private score: number;

	public constructor (pseudo: String, cote: number) {
		this.pseudo = pseudo;
		this.cote = cote;
		this.score = 0;
	}
	
	public abstract jouer(plateau: Plateau): number;
	
	public ajouterPoint(point: number) {
		this.score = (this.score + point);
	}
	
	public getScore(): number {
		return this.score;
	}
	
	public getPseudo(): String {
		return this.pseudo;
	}
	
	public setPseudo(pseudo: String) {
		this.pseudo = this.pseudo;
	}
	
	public setCote(cote: number) {
		this.cote = this.cote;
	}
	
	public afficher(): String {
		return "Le joueur " + this.pseudo + " a "+ this.score + " points.";
	}
}